mpp/include/mpp_do_update_ad.fh

```diff
@@ -111,9 +111,10 @@
                   msgsize = msgsize + (ie-is+1)*(je-js+1)
                end if
             end do
-            call mpp_send( msgsize, plen=1, to_pe=overPtr%pe, tag=COMM_TAG_1 )
+            l = overPtr%pe - mpp_root_pe()
+            msg3(l) = msgsize
          enddo
-         call mpp_sync_self(check=EVENT_RECV)
+         call mpp_alltoall(msg3, 1, msg1, 1)
```

Comment:
Can you add a comment on why you switched from mpp_sync_self to mpp_alltoall

Reply:
It would be good to note that all-to-all may have scaling issues on very large systems or require some MPI tuning parameters.

Reply:
added comment lines accordingly